src/database/statistics/spreadUtil.ts

```diff
@@ -127,3 +127,54 @@ export function getSpreadDates(
 
   return dates;
 }
+
+/**
+ * Helper function to eliminate duplicate weeks, sprints or months
+ * because the calculated intervals per developer may differ
+ * and would have been taken as separate intervals into account,
+ * altough it was the same interval of commiting.
+ * @param timeSpreadPairs An object with timestamps as keys, and an object with developer:sprad pairs as a value
+ * @param days The days, the interval blongs to, i.e. 7 (weeks), 14 (sprint), 30(month)
+ * @returns
+ */
+export function rearangeTimeslots(
```

Comment:
This method doesn't make much sense to me. You explained in the `devSpreadRepo`:
> Now, we need a little fix; consider dev A, which has contributed in week X, which starts at '2021-09-06' and then in week Y which starts at '2021-09-13' again, because his timeslots were calculated with his daily commit timestamps like this.  
Now, dev B has his own time interval computation based on his daily commit timestamps, consider he has contributed in week Z which starts at '2021-09-09'.    
So, we don't want to count the weeks as 3 weeks, as the week Z actually belongs in week X, we want to have 2 weeks of them and add dev B to the week X and delete week Z.  
This happens in rearangeTimeSlots() for weeks, sprints and months, as days are already precisely.

The `getSpreadsForDev` method indexes weeks using the calendar week. So week X and week Z would be mapped to the same index for devs A and B. What exactly do you rearrange or drop?

Reply:
Also, it should be `rearrangeTimeslots`

Reply:
Sorry that you had to read this, this is an old method which has no more functionality. I just outsourced this function to delete it in the next PR I will create.